plugin/tools.py

```diff
@@ -452,3 +482,16 @@ def get_clang_version_str(clang_binary):
         else:
             raise RuntimeError(
                 " Couldn't find clang version in clang version output.")
+
+    @staticmethod
+    def get_unique_str(init_string):
+        """ Generate md5 unique sting hash given init_string """
+        import hashlib
+        return hashlib.md5(init_string.encode('utf-8')).hexdigest()
+
+    @staticmethod
+    def get_position_hash(view, position_in_file):
+        """ Generate md5 unique sting hash from position in the file """
+        unique_id = Tools.get_unique_str(
+            view.file_name() + str(position_in_file))
```

Comment:
On the other hand, to disambiguate the string i need full name of the file which is unnecessarily long. Comparing these names is not as nice as hashes. Outputting the full name will anyway not be too helpful. So now I believe, I've changed my mind again :) Hashes look good to me now.